benchexec/tools/metaval.py

```diff
@@ -0,0 +1,89 @@
+"""
+BenchExec is a framework for reliable benchmarking.
+This file is part of BenchExec.
+
+Copyright (C) 2007-2019  Dirk Beyer
+All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import benchexec.util as util
+import benchexec.tools.template
+import benchexec.result as result
+import benchexec.tools.cpachecker as cpachecker
+import os
+
+
+class Tool(benchexec.tools.template.BaseTool):
+
+    REQUIRED_PATHS = ["CPAchecker", "esbmc"]
+    TOOL_TO_PATH_MAP = {"cpachecker": ["CPAchecker", "scripts"], "esbmc": ["esbmc"]}
```

Comment:
Why is `scripts` here? This should be specific knowledge of the wrapped tool-info module, shouldn't it?

Reply:
because finding the executable via the utils method did not work without it for some reason. I have no time to debug this *now*. Might do it later

Reply:
Ok, I retried it without the "scripts", it somehow suddenly works without, as I would have expected, so it can be removed. It is however not knowledge specific to the wrapped tool-info module. The wrapped tool is in a sub-folder in metaval's directory, and I control the name of that folder. So instead of "CPAchecker" I could call it "CPAchecker_1.8" , have multiple such directories/versions etc.

Reply:
> Ok, I retried it without the "scripts", it somehow suddenly works without,

Looks like I retried this in the wrong benchexec folder on my PC. The `CPAchecker/scripts` was necessary because though the tool info module of CPAchecker has `scripts/cpa.sh` as fallback for searching the executable, this is only tried in the current working directory. But the current working directory for my tool is not the CPAchecker folder, it is one above:
```
/metaval
├── CPAchecker
├── esbmc
├── LICENSE
└── metaval.sh
```

I changed the tool mechanics to enter the wrapped tool's folder now before resolving everything, such that i can fully delegate everything to the wrapped tool-info module.